src/govuk/components/character-count/character-count.mjs

```diff
@@ -96,9 +96,9 @@ function CharacterCount ($module, config) {
   })
 
   // Determine the limit attribute (characters or words)
-  if (this.config.maxwords) {
+  if ('maxwords' in this.config && this.config.maxwords) {
```

Comment:
Optional chaining (`this.config?.maxwords || this.config?maxlength`) would make the code work (provided we accept that a `0` value mean the property is not set, which is kind of reasonable when counting characters).

However Typescript would choke on it with the current types. 

`this.config` is being typed as `CharacterCountConfigWithMaxLength | CharacterCountConfigWithMaxWords` (via `CharacterCountConfig`). This allows TypeScript to check that the provided config provides either `maxlength` or `maxwords` but not both (it can provide neither. 

The check for `'xxx' in this.config` allows TypeScript to be sure that when we access the `maxlength` or `maxwords` inside the if we manipulate the right type of config. Within `if('maxlength' in this.config)` you'd be warned if you tried `this.config.maxwords` for example, as the `if` makes TypeScript consider `this.config` a `CharacterCountConfigWithMaxLength` (so without a `maxwords` property).

Reply:
Might be clearer to explain with an example?

Knowing we have two types of config:
https://govuk-frontend-review.herokuapp.com/docs/javascript/global.html#CharacterCountConfig

1. **CharacterCountConfigWithMaxLength**
2. **CharacterCountConfigWithMaxWords**

Plus we're quite strict, you can only use one or the other

Remember JavaScript has "falsy values" and we use `{number}` type 😬 

```mjs
const config1 = { maxlength: 0 }
const config2 = { maxwords: 0 }
```

Here's the problem! How do you tell which type of config we have?

```mjs
// These checks are all false 😱 
!config1.maxlength
!config2.maxwords
!config1.maxwords
!config2.maxlength
```

You have to add the `in` operator to confirm:

```mjs
// These checks are all true ✅
'maxlength' in config1.
'maxwords' in config2

// These checks are all false ❌
'maxwords' in config1
'maxlength' in config2
```

Combine them together and you can check for existence _and_ "truthy"-ness